src/components/button/button.tsx

```diff
@@ -42,38 +34,51 @@ interface CommonButtonProps {
   className?: string
 }
 
-interface ButtonAsButtonElementProps extends CommonButtonProps, ButtonHTMLAttributes<HTMLButtonElement> {
-  href?: never
-  target?: never
-  disabled?: boolean
-  onClick?: MouseEventHandler<HTMLButtonElement>
+// Define a specialized type for tertiary variant for prop removePadding
+type TertiaryButtonProps = {
+  variant: 'tertiary'
+  removePadding?: boolean // Only tertiary buttons can use removePadding
 }
 
-interface ButtonAsAnchorElementProps extends CommonButtonProps, AnchorHTMLAttributes<HTMLAnchorElement> {
-  variant?: ButtonAsAnchorVariant
-  /** Button styled <a> element should always have href */
-  href: string
-  target?: string
-  rel?: string
-  /** Anchor elements cannot be disabled. Use a button element if the component needs to be in a disabled state */
-  disabled?: never
-  onClick?: MouseEventHandler<HTMLAnchorElement>
+// Define the general type for non-tertiary variants
+type NonTertiaryButtonProps = {
+  variant?: Exclude<ButtonAsButtonVariant, 'tertiary'>
+  removePadding?: never // Disallow removePadding for other variants
 }
 
+// Combine the above two into a Variant-specific prop type
+type VariantSpecificProps = TertiaryButtonProps | NonTertiaryButtonProps
+
+type ButtonAsButtonElementProps = CommonButtonProps &
+  VariantSpecificProps &
+  ButtonHTMLAttributes<HTMLButtonElement> & {
+    href?: never
+    target?: never
+    disabled?: boolean
+    onClick?: MouseEventHandler<HTMLButtonElement>
+  }
+
+type ButtonAsAnchorElementProps = CommonButtonProps &
+  VariantSpecificProps &
+  AnchorHTMLAttributes<HTMLAnchorElement> & {
+    variant?: ButtonAsAnchorVariant
+    /** Button styled <a> element should always have href */
+    href: string
+    target?: string
+    rel?: string
+    /** Anchor elements cannot be disabled. Use a button element if the component needs to be in a disabled state */
+    disabled?: never
+    onClick?: MouseEventHandler<HTMLAnchorElement>
+  }
+
```

Comment:
**note:** I don't love how complex these types are becoming. Not really sure what our options are to simplify them, just flagging that this level of complexity can lead to poor DX as type errors become hard to read 🤔 

Intersections are also not the best way of combining types. See the [(draft) prop interface guidelines](https://consolegroup.atlassian.net/wiki/spaces/DP/pages/3127509024/Prop+Interface+Guidelines#Prefer-interface-over-intersections-when-composing-types) for details on why interface/extends is preferred over intersections. No need to rewrite these types yet; rather I think we should do a sweep early next year to tidy up some of our component prop types (there's TS config changes I'd like to rollout, along with ES Lint upgrades etc).